test/forms/DeclarationFormMappingSpec.scala

```diff
@@ -1026,4 +1026,262 @@ class DeclarationFormMappingSpec extends WordSpec
       }
     }
   }
+  "Date" should {
+
+    "bind" when {
+
+      "valid values are bound" in {
+
+        forAll { date: Date =>
+
+          Form(dateMapping).fillAndValidate(date).fold(
+            error => fail(s"Failed with errors:\n${error.errors.map(_.message).mkString("\n")}"),
+            result => result mustBe date
+          )
+        }
+      }
+    }
+
+    "fail" when {
+
+      "Invalid day is entered" in {
+
+        forAll(arbitrary[Date], intBetweenRange(35, 60)) { (date, day) =>
+
+          Form(dateMapping).fillAndValidate(date.copy(day = day)).fold(
+            _.error("day") must haveMessage("Day is invalid"),
+            _ => fail("Should not succeed")
+          )
+        }
+      }
+
+      "Invalid month is entered" in {
+
+        forAll(arbitrary[Date], intBetweenRange(13, 30)) { (date, month) =>
+
+          val invalidDate = date.copy(month = month)
+          Form(dateMapping).fillAndValidate(invalidDate).fold(
+            _.error("month") must haveMessage("Month is invalid"),
+            _ => fail("Should not succeed")
+          )
+        }
+      }
+
+      "Invalid year is entered" in {
+
+        forAll(arbitrary[Date], intBetweenRange(1890, 1899)) { (date, year) =>
+
+          val invalidDate = date.copy(year = year)
+          Form(dateMapping).fillAndValidate(invalidDate).fold(
+            _.error("year") must haveMessage("Year is invalid"),
+            _ => fail("Should not succeed")
+          )
+        }
+      }
+    }
+  }
+
+  "MeasureMapping" should {
+
+    "bind" when {
+
+      "valid values are passed" in {
+
+        forAll { measure: Measure =>
+
+          Form(measureMapping).fillAndValidate(measure).fold(
+            e => fail(s"form should not fail: ${e.errors}"),
+            success => success mustBe measure
+          )
+        }
+      }
+    }
+
+    "fail" when {
+
+      "unitCode is greater than 5" in {
+
+        val badData = stringsLongerThan(5)
```

Comment:
use minStringLength here as it will always generate a value

```suggestion
        val badData = minStringLength(6)
```

Reply:
updated